gems/smithy/lib/smithy/generators/client.rb

```diff
@@ -11,41 +11,67 @@ def initialize(plan)
         super
       end
 
-      # @return [Enumerator<String, String>] The file paths and their contents to generate.
       def generate
-        files = source_files
-        files.each do |file, content|
+        gem_files.each_with_object([]) do |(file, content), files|
           next if file == "lib/#{@gem_name}/customizations.rb" && should_skip_customizations?
 
           create_file file, content
+          files << file
         end
-        files
+      end
+
+      def source
```

Comment:
I think it would be worthwhile to document what `generate` and `source` methods does somewhere, since it is touching different set of files (gem_files vs source_files)

Reply:
They are documented at their entry point in the Smithy module.